src/core/Observable_stat.cpp

```diff
@@ -32,28 +32,15 @@
 
 extern boost::mpi::communicator comm_cart;
 
-/** Tracker of observables */
-std::vector<Observable_stat_wrapper *> &registered_observables() {
-  static std::vector<Observable_stat_wrapper *> s_registered_observables;
-  return s_registered_observables;
-}
-
-void Observable_stat_wrapper::register_obs() {
-  registered_observables().push_back(this);
-}
-
-void invalidate_obs() {
-  for (auto *obs : registered_observables())
-    obs->is_initialized = false;
-}
-
 void Observable_stat::resize() {
   // number of chunks for different interaction types
-  auto const n_coulomb =
-      m_pressure_obs ? Coulomb::pressure_n() : Coulomb::energy_n();
-  auto const n_dipolar =
-      m_pressure_obs ? Dipole::pressure_n() : Dipole::energy_n();
-  size_t n_vs = 0;
+  size_t n_coulomb = 0, n_dipolar = 0, n_vs = 0;
```

Comment:
I'd tend to make the contributions exist in any case, to reduce the noise level. Also those can be const either way.

Reply:
done